CIP-????/README.md

```diff
@@ -0,0 +1,250 @@
+
+- The question and answers
+- The index of the chosen answer from the available choices
+- A signature from the SPO's current VRF credentials or Ed25519 declared cold credentials.
+
+> **Note**
+> In this document, every time we refer to a _serialized object_, we refer to its **canonical** CBOR representation. In particular, keys in a map are always ordered alphabetically.
+
+### Question structure
+
+A question is posted in a transaction's metadata using the metadata label `94` and the following metadata structure:
+
+```cbor
+question =
+  { 0: prompt
+  , 1: [ * choice ]
+  , ? "_": nonce
+  }
+
+prompt =
+  [ * text .size (0..64) ]
+
+choice =
+  [ * text .size (0..64) ]
+
+nonce =
+  uint
+```
+
+A nonce is optionally included to provide non-replayability should the same question and answers be asked multiple times over different periods. The transaction carrying a question **must** be signed by one of the genesis delegate keys to be considered valid. This genesis key signature isn't captured in the metadata but in the transaction itself as an extra signatory.
+
+For example, in [CBOR-diagnostic](https://www.rfc-editor.org/rfc/rfc8949#name-diagnostic-notation) notation:
+
+```cbor
+{ 94:
+  { 0: "Pineapples on pizza?"
+  , 1:
+    [ "yes"
+    , "no"
+    ]
+  }
+}
+```
+
+Which serialises down to:
+
+```
+A1185EA2007450696E656170706C6573206F6E2070697A7A613F018263796573
+626E6F
+```
+
+### Answer structure
+
+Similarly, an answer to a question is posted as transaction's metadata using the label `94` and the following metadata structure:
+
+```cbor
+answer =
+  { 2: question_hash
+  , 3: choice
+  , ? 4: vrf_witness
+  , ? 5: ed25519_witness
+  }
+
+question_hash =
+  bytes .size 32
+
+vrf_witness =
+  [ $vrf_vkey, vrf_proof ]
+
+vrf_proof =
+  [ bytes .size 64, bytes .size 16 ]
+
+ed25519_witness =
+  [ $vkey, $signature ]
+```
+
+Some remarks:
+
+1. The field `2` (`question_hash`) is a blake2b-256 hash digest, whose preimage is the entire serialised question metadata payload (with the `94` top-level label).
+1. The fields `4` and `5` are not mutually optional; said differently, at least one of the two must be provided.
+1. The `$vrf_vkey`, `$vkey` and `$signature` are as defined in the original [Cardano ledger specifications](https://github.com/input-output-hk/cardano-ledger/blob/master/eras/babbage/test-suite/cddl-files/real/crypto.cddl) with the usual rules applying to transaction metadata (i.e. byte string longer than 64 bytes are encoded as lists)
+1. In either case (VRF or Ed25519), the witness' signature is calculated from the blake2b-256 hash digest of the serialised `answer` restricted to fields `2` and `3`.
+
+For example, in [CBOR-diagnostic](https://www.rfc-editor.org/rfc/rfc8949#name-diagnostic-notation) notation:
+
+```cbor
+{
+  94: {
+    2: h'3d4854aed3e31d0f5b6637fcd73e8713d9eee86a81d20598f6cb88d9b2bc0a39',
```

Comment:
I just wanted to confirm, is the hash specified here not the "BLAKE2B-256" digest of the serialized pineapple poll example?

If I hash it, I get `c9a077a1098d73498d17e9ea27045af820c311ced91f8c2bb9b5c7f446379063`.  Here's the code snippet that you can run with deno for reference:
```ts
import * as mod from "https://deno.land/std@0.182.0/crypto/mod.ts";
import {decode} from "https://deno.land/std/encoding/hex.ts";

const hexString="A1185EA2007450696E656170706C6573206F6E2070697A7A613F018263796573626E6F";

const hash = await mod.crypto.subtle.digest(
	"BLAKE2B-256",
	decode(new TextEncoder().encode(hexString))
);
console.log(mod.toHashString(hash));
```

Reply:
The examples are NOT test vectors indeed, mostly gibberish data. Could be arguably more useful if there were test vectors. I can try pulling up the test scenarios implemented for input-output-hk/cardano-node#5050

Reply:
They are now.